code/modules/hydroponics/hydroponics.dm

```diff
@@ -1243,6 +1268,17 @@
 	plant_health = add_output_port("Plant Health", PORT_TYPE_NUMBER)
 	reagents_level = add_output_port("Reagents Level", PORT_TYPE_NUMBER)
 
+/obj/machinery/hydroponics/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	. = ..()
+	if(!istype(arrived, /obj/item/clothing/head/mob_holder/snail))
+		return
+	our_snail = new
+	vis_contents += our_snail
+	our_snail.layer = layer + 0.01
+	var/obj/item/clothing/head/mob_holder/snail = arrived
+	RegisterSignal(snail.held_mob, COMSIG_LIVING_DEATH, PROC_REF(on_snail_death)) //rip
```

Comment:
You should register `COMSIG_ATOM_QDELETING` here as well in case something destroys the hat without technically killing the snail
You also don't actually unregister this signal if the snail exits the tray